src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -50,8 +50,6 @@
       Text="RuntimeIdentifier is required for native compilation. Try running dotnet publish with the -r option value specified." />
     <Error Condition="'$(GeneratePackageOnBuild)' == 'true'" Text="GeneratePackageOnBuild is not supported for native compilation." />
     <Error Condition="'$(OutputType)' != 'Library' and '$(NativeLib)' != '' and '$(CustomNativeMain)' != 'true'" Text="NativeLib requires OutputType=Library." />
-    <!-- See https://github.com/dotnet/runtime/issues/89346 for details -->
-    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />
 
```

Comment:
```suggestion
    <!-- See https://github.com/dotnet/runtime/issues/89346 for details -->
    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true' and '$(_SuppressNativeLibEventSourceError)' != 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />

```

Reply:
Is this the correct issue to link to?

Do we want to make this self-serviceable? There's no hint that this is actually possible to enable.

I'm surprised we went from "Event source support in native libraries is so rare we can ship 8.0 with this completely unsupported" to "What if two libraries do this at the same time" and optimize for the latter.

Reply:
I think of the block as more 'do not leave the user in a bad/undefined state by default' rather than a call on how common a scenario is.

Self-serviceability seems more tied to whether we expect/want users to use this. If no, then I think the undocumented property makes sense (similar to trimming + WPF). If yes, then I think the bypass property should get a better name and be documented and the error message should call out the property and link to docs.

Reply:
IMHO this multiple event pipe problem is not specific to shared libraries. It can also happen if someone turns on EventSourceSupport in an EXE and then starts calling CoreCLR hosting apis to load CoreCLR.

I already know of at least two people/teams within Microsoft that do this. I don't think they enable EventSourceSupport but if they did, they'd run into the same problem. We don't block EventSourceSupport in EXEs because someone might load something else with event pipe IPC into the process.

Reply:
If we want to do something about this, the something I'd do is to define the failure mode for that.